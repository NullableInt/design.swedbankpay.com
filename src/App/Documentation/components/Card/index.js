import React, { Component } from "react";

import ComponentPreview from "../../utils/ComponentPreview";
import DocToc from "../../utils/DocToc";
import CardComponent from "../../../components/Card";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Overview... <code className="token property">.card</code>...</p>
        <p>... <code className="token property">.card-default</code>...</p>
        <p>... <code className="token property">.card-brand</code>...</p>
        <p>... <code className="token property">.card-light</code>...</p>
        <ComponentPreview language="html" showCasePanel={true} codeFigure={true}>
            <div className="cards">
                <ul>
                    <CardComponent
                        type="default"
                        title="Card default"
                        imgSrc="https://picsum.photos/300"
                        icon="account_balance"
                        highlightTitle="725 NOK"
                        highlightSubTitle="Outstanding balance"
                        text="This is the cards text"
                        smallText="And the small text"
                        btnTxt="Button"
                        footerTxt="This is some text that goes in the footer."
                        footerLinkTxt="You wanna click this link?"
                    />
                    <CardComponent
                        type="brand"
                        title="Card brand"
                        imgSrc="https://picsum.photos/300"
                        icon="face"
                        highlightTitle="Bob Corlsan"
                        highlightSubTitle="bob.corlsan@payex.com"
                        text="This is the cards text"
                        smallText="And the small text"
                        btnTxt="Button"
                        footerTxt="This is some text that goes in the footer."
                        footerLinkTxt="You wanna click this link?"
                    />
                    <CardComponent
                        type="light"
                        title="Card light"
                        imgSrc="https://picsum.photos/300"
                        icon="verified_user"
                        highlightTitle="75%"
                        highlightSubTitle="Complete"
                        text="This is the cards text"
                        smallText="And the small text"
                        btnTxt="Button"
                        footerTxt="This is some text that goes in the footer."
                        footerLinkTxt="You wanna click this link?"
                    />
                </ul>
            </div>
        </ComponentPreview>
    </>
);

const CardText = () => (
    <div className="col-md-12 col-lg-10 doc-body">
        <div className="alert alert-danger in">
            This component is under development and will probably change.
        </div>
        <p className="lead">Card...</p>
        <Overview />
    </div>
);

class Card extends Component {
    render () {
        return (
            <div className="doc-container">
                <div className="row">
                    <CardText />
                    <DocToc component={CardText} />
                </div>
            </div>
        );
    }
}

export default Card;

/* for testing */
export { Overview, CardText };
