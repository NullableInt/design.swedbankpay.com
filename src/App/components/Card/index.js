import React from "react";
import PropTypes from "prop-types";

const Card = ({ type, title, imgSrc, text, textSection, smallText, btn, btnClass, bodyClass, footerTxt, footerLink, footerLinkTxt, children }) => (
    <div className={`card${type ? ` card-${type}` : ""}`}>
        {imgSrc ? <div className="card-img">{"\n"}
            <img src={imgSrc} alt=""/>{"\n"}
        </div> : null}
        {title ?
            <header>
                {title}
            </header>
            : null}
        <div className={`card-body ${bodyClass ? ` ${bodyClass}` : ""}`}>
            {children ? children : null }
            {text ? <p>{text}</p> : null}{"\n"}
            {textSection ?
                textSection.map((secText, i) => <p key={`card-p-${i}`}>{secText}</p>)
                : null
            }
            {btn ? <button type="button" className={`btn btn-executive${btnClass ? ` ${btnClass}` : ""}`}>Button</button> : null}
        </div>
        {footerTxt || footerLinkTxt ? <footer>
            {smallText ? <small>{smallText}</small> : null}
            {footerTxt ? <div className="footer-text"><p>{footerTxt}</p></div> : null}
            {footerLinkTxt ? <div className="footer-link"><a href={footerLink || "#"}>{footerLinkTxt}</a></div> : null}
        </footer> : null}
    </div>
);

Card.propTypes = {
    type: PropTypes.oneOf(["primary", "secondary", "plain"]),
    imgSrc: PropTypes.string,
    icon: PropTypes.string,
    text: PropTypes.string,
    smallText: PropTypes.string,
    textSection: PropTypes.array,
    btn: PropTypes.bool,
    footerTxt: PropTypes.string,
    footerLink: PropTypes.string,
    footerLinkTxt: PropTypes.string
};

export default Card;
