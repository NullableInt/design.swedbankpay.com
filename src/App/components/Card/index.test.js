import React from "react";
import { shallow } from "enzyme";

import Card from "./index";

describe("Component: Card - ", () => {
    it("is defined", () => {
        expect(Card).toBeDefined();
    });

    it("throws an error if type does not match proptypes", () => {
        console.error = jest.fn();

        const wrapper = shallow(<Card type="error" />);

        expect(console.error).toHaveBeenCalled();
        expect(wrapper).toMatchSnapshot();
    });

    it("renders with class name card if no type is specified", () => {
        const wrapper = shallow(<Card />);

        expect(wrapper.find(".card")).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders a header", () => {
        const wrapper = shallow(<Card title="Card default"/>);

        expect(wrapper).toMatchSnapshot();
    });

    it("renders an image", () => {
        const wrapper = shallow(<Card imgSrc="url.address" />);

        expect(wrapper.find(".card-img")).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders an icon", () => {
        const wrapper = shallow(<Card icon="home" />);

        expect(wrapper.find(".material-icons")).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders a highlight title", () => {
        const wrapper = shallow(<Card highlightTitle="Balance" />);

        expect(wrapper.find(".highlight-title")).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders a highlight subtitle", () => {
        const wrapper = shallow(<Card highlightSubTitle="remaining amount to pay" />);

        expect(wrapper.find(".highlight-subtitle")).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders footer text", () => {
        const wrapper = shallow(<Card footerTxt="Footer information" />);

        expect(wrapper.find(".footer-text")).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders footer link text", () => {
        const wrapper = shallow(<Card footerLinkTxt="Click this footer link" />);

        expect(wrapper.find(".footer-link")).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });
});
