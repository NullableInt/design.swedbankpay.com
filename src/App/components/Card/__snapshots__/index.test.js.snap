// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Card -  renders a child object 1`] = `
<div
  className="card"
>
  <div
    className="card-body "
  >
    This is child text
    

  </div>
</div>
`;

exports[`Component: Card -  renders a text in card-body 1`] = `
<div
  className="card"
>
  <div
    className="card-body "
  >
    <p>
      Text that goes in card body
    </p>
    

  </div>
</div>
`;

exports[`Component: Card -  renders a title 1`] = `
<div
  className="card"
>
  <header>
    Card title goes here
  </header>
  <div
    className="card-body "
  >
    

  </div>
</div>
`;

exports[`Component: Card -  renders an image 1`] = `
<div
  className="card"
>
  <div
    className="card-img"
  >
    

    <img
      alt=""
      src="url.address"
    />
    

  </div>
  <div
    className="card-body "
  >
    

  </div>
</div>
`;

exports[`Component: Card -  renders footer link text 1`] = `
<div
  className="card"
>
  <div
    className="card-body "
  >
    

  </div>
  <footer>
    <div
      className="footer-link"
    >
      <a
        href="#"
      >
        Click this footer link
      </a>
    </div>
  </footer>
</div>
`;

exports[`Component: Card -  renders footer text 1`] = `
<div
  className="card"
>
  <div
    className="card-body "
  >
    

  </div>
  <footer>
    <div
      className="footer-text"
    >
      <p>
        Footer information
      </p>
    </div>
  </footer>
</div>
`;

exports[`Component: Card -  renders with a button 1`] = `
<div
  className="card"
>
  <div
    className="card-body "
  >
    

    <button
      className="btn btn-executive"
      type="button"
    >
      Button
    </button>
  </div>
</div>
`;

exports[`Component: Card -  renders with class name card 1`] = `
<div
  className="card"
>
  <div
    className="card-body "
  >
    

  </div>
</div>
`;

exports[`Component: Card -  renders with correct class name 1`] = `
<div
  className="card card-brand"
>
  <div
    className="card-body "
  >
    

  </div>
</div>
`;

exports[`Component: Card -  renders with only body when no props are passed 1`] = `
"<div class=\\"card\\"><div class=\\"card-body \\">
</div></div>"
`;

exports[`Component: Card -  throws an error if type does not match proptypes 1`] = `
<div
  className="card card-error"
>
  <div
    className="card-body "
  >
    

  </div>
</div>
`;
